### PR TITLE
Feature/tables.synchronized table 2156

### PR DESCRIPTION
**Pull request by mostlymagic**
Had deleted previous fork by accident.
Suggested changes implemented in this one.

### CODE REVIEW

File: guava-gwt/src-super/com/google/common/collect/super/com/google/common/collect/Synchronized.java [hunk 1]
```diff
@@ -1303,5 +1303,192 @@ public E remove() {
...
+      synchronized (mutex) {
+        return set(delegate().columnKeySet(), mutex);
+      }
+    }
+
+    @Override
+    public Map<C, Map<R, V>> columnMap() {
+      synchronized (mutex) {
+        Function<Map<R, V>, Map<R, V>> mapFunction = synchronizedMapWrapper(mutex);
+          Map<C, Map<R, V>> columnMap = delegate().columnMap();
```
Comment (lowasser) [inline guava-gwt/src-super/com/google/common/collect/super/com/google/common/collect/Synchronized.java]: Wait, I wasn't referring to indenting these; they should be at the same level as the rest of the code...

File: guava-tests/test/com/google/common/collect/SynchronizedTableTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        @Override
+        public int hashCode() {
+            assertTrue(Thread.holdsLock(mutex));
+            return super.hashCode();
+        }
+
+        private static long serialVersionUID = 0;
+    }
+
+  /*
```
Comment (lowasser) [inline guava-tests/test/com/google/common/collect/SynchronizedTableTest.java]: I think it's this comment that should be..indented +2?  The leading / of the comment should be at the same level as the method signatures.

Comment (mostlymagic) [inline guava-tests/test/com/google/common/collect/SynchronizedTableTest.java]: seems to have been an IDE settings mixup. will provide a fix, sorry

**Comment by shapowang** [comment]
Does anyone resolve the conflict and merge the code?
I cannot find the `Tables.synchronizedTable` function in guava 20.0-rc1.This function would be useful,please.

**Comment by cpovirk** [comment]
I'm not sure why we ended up with our own implementation instead of this one, but we ended up adding one in https://github.com/google/guava/commit/e3f9ba6a707b03559910bceffb74244ac9a8db13

**Pull request closed without merging**